cmd/operator/main.go

```diff
@@ -155,11 +157,30 @@ func runDaemon(ctx context.Context, logger *logrus.Logger, cfg *config.Config) e
 			logger.Warn("no service matches the targets")
 		}
 
-		// TODO: Handle all the filtered services
-		err = handleRollout(ctx, logger, services[0], cfg.Strategy)
-		if err != nil {
-			return errors.Wrap(err, "error when handling rollout")
+		var (
+			errs []error
+			wg   sync.WaitGroup
+		)
+		for _, service := range services {
+			wg.Add(1)
+			go func(ctx context.Context, lg *logrus.Logger, svc *rollout.ServiceRecord, strategy *config.Strategy) {
+				err = handleRollout(ctx, lg, svc, strategy)
+				if err != nil {
+					errs = append(errs, err)
```

Comment:
It seems unsafe to simply append to a global slice from inside the go func. I'd expect a RW mutex.

Reply:
Oohhh, true. Good catch!